main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  # my-each
+
+  def my_each
+    ind = 0
+
+    while ind != length
+      yield self[ind]
+      ind += 1
+    end
+
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    index = 0
+    my_each do |el|
+      yield(el, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select
+
+  def my_select
+    filtered_arr = []
+
+    while length != 0
+      yield(first) && filtered_arr.push(first)
+      shift
+    end
+
+    filtered_arr
+  end
```

Comment:
- [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Your method is not returning an enumerator when no block is given.
- It should return an array with all the elements that meet the given condition (it's returning more).
- It should work with ranges.
- It shouldn't mutate the original array.